File: Leetcode/938.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int rangeSumBST(TreeNode* root, int low, int high) {
        if (!root) return 0;
        int temp = (root->val >= low && root->val <= high) ? (root->val) : 0;
        int leftAns = (root->left) ? rangeSumBST(root->left, low, high) : 0;
        int rightAns = (root->right) ? rangeSumBST(root->right, low, high) : 0;
        return temp + leftAns + rightAns;
    }
};

File: Leetcode/938/938.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func dfs(node *TreeNode, low int, high int, sum *int) {
	if node != nil {
		if node.Val < low {
			dfs(node.Right, low, high, sum)
		} else if node.Val > high {
			dfs(node.Left, low, high, sum)
		} else {
			*sum += node.Val
			dfs(node.Left, low, high, sum)
			dfs(node.Right, low, high, sum)
		}
	}
}

func rangeSumBST(root *TreeNode, low int, high int) int {
	sum := 0
	dfs(root, low, high, &sum)
	return sum
}